src/Type/Php/ArrayLastDynamicReturnTypeExtension.php

```diff
@@ -0,0 +1,48 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Type\Php;
+
+use PhpParser\Node\Expr\FuncCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\DependencyInjection\AutowiredService;
+use PHPStan\Reflection\FunctionReflection;
+use PHPStan\Type\DynamicFunctionReturnTypeExtension;
+use PHPStan\Type\NullType;
+use PHPStan\Type\Type;
+use PHPStan\Type\TypeCombinator;
+use function count;
+
+#[AutowiredService]
+final class ArrayLastDynamicReturnTypeExtension implements DynamicFunctionReturnTypeExtension
```

Comment:
as this 2 extensions are very similar, you might consider using a single extension class for handling both

Reply:
With the recent refactoring they are now merged into one